### DISCUSSION TITLE
功能请求-关于图片

### DISCUSSION THREAD

**Discussion by idler8** [AlistGo/alist#3201] [Ideas]
主要：文件缩略图生成支持（视频文件没有缩略图无法区分，使用缩略图可以优化流量）
次要：苹果HEIC图片前端页面支持（我使用手机WEBDAV软件连接，图片正常能显示，WEB前端上建议增加）
次要：苹果手机拍摄的MOV文件，在WEB上黑屏

**Comment by idler8** [comment]
或者这个Webdav本机储存的缩略图读取逻辑是怎么样的？我可以弄个其它项目，挂在旁边定时生成

**Comment by idler8** [comment]
https://github.com/alist-org/alist/blob/9d9377f65d4aae539a8440d7440b794f452ae5bd/drivers/local/driver.go#L71
此处增加一个自定义的预览图片加载目录就好了
可以在配置里设置某些文件扩展，使用某个目录下的文件为缩略图

**Reply by kokojacket** [reply]
请问这个怎么操作

**Comment by KnIfER** [comment]
你写个接口，thumb中返回一个插件算出来的url。我解析这个url，返回缩略图。

alist是否存在这样的接口？
